src/query/sql/src/planner/optimizer/s_expr.rs

```diff
@@ -320,5 +502,99 @@ fn find_subquery_in_expr(expr: &ScalarExpr) -> bool {
         ScalarExpr::CastExpr(expr) => find_subquery_in_expr(&expr.argument),
         ScalarExpr::SubqueryExpr(_) => true,
         ScalarExpr::UDFServerCall(expr) => expr.arguments.iter().any(find_subquery_in_expr),
+        ScalarExpr::UDFLambdaCall(expr) => find_subquery_in_expr(&expr.scalar),
+    }
+}
+
+pub fn get_udf_names(scalar: &ScalarExpr) -> Result<HashSet<String>> {
```

Comment:
Recommends to use `Visitor` for `ScalarExpr`. Here is an [example](https://github.com/datafuselabs/databend/blob/3f4e629f8ba38b030b6a9fa3fc84def4f90a61e9/src/query/sql/src/planner/binder/delete.rs#L187C1-L199)

Reply:
Fixed. Please review the newest commit.